src/tools/perf/perftest_daemon.c

```diff
@@ -247,14 +257,25 @@ ucp_perf_daemon_memh_import(ucp_perf_daemon_context_t *ctx, void *packed_memh)
 
     params.field_mask           = UCP_MEM_MAP_PARAM_FIELD_EXPORTED_MEMH_BUFFER;
     params.exported_memh_buffer = packed_memh;
-    status                      = ucp_mem_map(ctx->context, &params, &memh);
+
+    status = ucp_mem_map(ctx->context, &params, &memh);
     if (status != UCS_OK) {
-        ucs_fatal("failed to import memory (%s)", ucs_status_string(status));
+        ucs_error("failed to import memory (%s)", ucs_status_string(status));
+        return NULL;
     }
 
     return memh;
 }
 
+/*
+ * From the given buffer unpack an array of bytes.
```

Comment:
Unpack array length and array contents from a buffer.

rename the function to ucp_perf_unpack_array
maybe return ucs_array_t?

Reply:
Ok for documentation and rename.

For `ucs_array_t`- I hesitate a bit, because it will also require to make symmetric change in the "pack" counterpart (which will be part of batch commit 3). Currently on the host side we have `void *, size_t` pair representing a memory block, which we get from the existing function:
```
ucp_memh_pack(ucp_mem_h memh, const ucp_memh_pack_params_t *params,
              void **buffer_p, size_t *buffer_size_p)
```
And the result of that function we store inside the new structure:
```
typedef struct {
    void   *address;
    size_t length;
} ucx_perf_exported_mem_t;
```
So yes, technically this is an array. And if we want to replace it with `ucs_array`, then the code will be a bit more verbose:
```
UCS_ARRAY_DECLARE_TYPE(ucs_byte_array, uint64_t, uint8_t);

static void
ucp_perf_unpack_byte_array(void **ptr, ucs_byte_array *array)
{
    array->length = *ucs_serialize_next(ptr, uint64_t);
    array->buffer = (length != 0) ?
                        ucs_serialize_next_raw(ptr, void, array->length) : NULL;
    array->capacity = array->length;
    array->is_fixed = 1;
}

static void
ucp_perf_pack_byte_array(void **ptr, const ucs_byte_array *array)
{
    *ucs_serialize_next(ptr, uint64_t) = array->length;
    if (array->length > 0) {
        memcpy(ucs_serialize_next_raw(ptr, void, array->length),
               array->buffer, array->length);
    }
}

static size_t ucp_perf_pack_daemon_info(const ucx_perf_context_t *perf,
                                        void *buffer, size_t buffer_size,
                                        const struct sockaddr_storage *address)
{
    void *p               = buffer;
    size_t address_length = (address != NULL) ? sizeof(*address) : 0;
    ucs_byte_array addr_bytes = UCS_ARRAY_FIXED_INITIALIZER(address, address_length);

    ucp_perf_pack_byte_array(&p, &addr_bytes);
    ucp_perf_pack_byte_array(&p, &perf->ucp.send_exported_mem);
    ucp_perf_pack_byte_array(&p, &perf->ucp.recv_exported_mem);
```
So in the pack side we need to perform conversion from `void *, size_t` into `byte_array` in 2 places.
Plus we need to store one extra field for `capacity`.
That's why I'm not 100% sure about this change, but I'm not against it either

If you like this approach - please let me know, I'll put in in place